map/mwm_url.hpp

```diff
@@ -21,11 +21,11 @@ namespace url_scheme
 {
 struct ApiPoint
 {
-  double m_lat;
-  double m_lon;
-  std::string m_name;
-  std::string m_id;
-  std::string m_style;
+  double m_lat = 0.0;
+  double m_lon = 0.0;
+  std::string m_name = "";
+  std::string m_id = "";
+  std::string m_style = "";
```

Comment:
What warning is here? I can understand lat, lon, but std::string m_name = ""; looks strange.

Reply:
This looks very strange. Default initialization should be enough. There should be another way to fix it (or the warning is false-positive).

Reply:
When I remove default initializers from the struct.

```
In file included from /home/runner/work/organicmaps/organicmaps/build/map/CMakeFiles/map.dir/Unity/unity_0_cxx.cxx:37:
/home/runner/work/organicmaps/organicmaps/map/mwm_url.cpp: In member function ‘void url_scheme::ParsedMapApi::ParseMapParam(const std::string&, const std::string&, std::vector<url_scheme::ApiPoint>&, bool&)’:
/home/runner/work/organicmaps/organicmaps/map/mwm_url.cpp:212:21: warning: missing initializer for member ‘url_scheme::ApiPoint::m_name’ [-Wmissing-field-initializers]
  212 |     points.push_back({.m_lat = lat, .m_lon = lon});
      |     ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/organicmaps/organicmaps/map/mwm_url.cpp:212:21: warning: missing initializer for member ‘url_scheme::ApiPoint::m_id’ [-Wmissing-field-initializers]
/home/runner/work/organicmaps/organicmaps/map/mwm_url.cpp:212:21: warning: missing initializer for member ‘url_scheme::ApiPoint::m_style’ [-Wmissing-field-initializers]
```

I think we can ignore this warning right now.

Reply:
Can you locally silence it?

Reply:
Is this warning also happens when your write points.push_back({lat, lon}); without var specifiers?

Reply:
Still exists:
```
In file included from /home/runner/work/organicmaps/organicmaps/build/map/CMakeFiles/map.dir/Unity/unity_0_cxx.cxx:37:
/home/runner/work/organicmaps/organicmaps/map/mwm_url.cpp: In member function ‘void url_scheme::ParsedMapApi::ParseMapParam(const std::string&, const std::string&, std::vector<url_scheme::ApiPoint>&, bool&)’:
/home/runner/work/organicmaps/organicmaps/map/mwm_url.cpp:212:21: warning: missing initializer for member ‘url_scheme::ApiPoint::m_name’ [-Wmissing-field-initializers]
  212 |     points.push_back({lat, lon});
      |     ~~~~~~~~~~~~~~~~^~~~~~~~~~~~
/home/runner/work/organicmaps/organicmaps/map/mwm_url.cpp:212:21: warning: missing initializer for member ‘url_scheme::ApiPoint::m_id’ [-Wmissing-field-initializers]
/home/runner/work/organicmaps/organicmaps/map/mwm_url.cpp:212:21: warning: missing initializer for member ‘url_scheme::ApiPoint::m_style’ [-Wmissing-field-initializers]
```

Reply:
This is a well-known controversial warning. Let's disable it here locally for gcc-only using ifdef and pragma.

Reply:
Hm, we should have a lot of warnings like this in this case. I'd prefer to not bother about it and don't put any pragmas garbage in the code.

Reply:
Non-silenced warnings will hide some important warnings. Explicit gcc silencing will help to make it clearer in the future. There are not so many similar warnings in the code, right?

Reply:
> Hm, we should have a lot of warnings like this in this case

>There are not so many similar warnings in the code, right?

It appears only here.

>I'd prefer to not bother about it and don't put any pragmas garbage in the code.

I agree. I don't like these pragmas.

> Non-silenced warnings will hide some important warnings

There are only 2-3 warnings left. Nothing will be hidden. I'd prefer to leave this warning for now and come back to it later. Maybe during the migration to C++20